### ISSUE TITLE
[Feature Request] Support the system window frame for the Voice Chat window

### ISSUE THREAD

**Issue by xmha97** [telegramdesktop/tdesktop#10002]
Please support the system window frame for the Voice Chat window.
![Telegram](https://user-images.githubusercontent.com/23485114/103111923-ef47ca00-4666-11eb-9f53-133e84e4e818.png)

**Comment by Demetri0** [comment]
Also for 1-1 dialog, please.

**Comment by andrei-a-papou** [comment]
I second that. Right now on Ubuntu 20.04 LTS the call window's titlebar looks really out of place... Thankfully, the main window's decorations look fine.

@ilya-fedin Could you comment on why the _Use system window frame_ setting doesn't apply to call windows? This looks inconsistent in terms of UX.

**Comment by ilya-fedin** [comment]
> Could you comment on why the _Use system window frame_ setting doesn't apply to call windows?

Because @john-preston doesn't want to provide that option for these windows

**Comment by andrei-a-papou** [comment]
@ilya-fedin I'm not sure I follow the logic here... There's an option that says _Use system window frame_. Clearly when a user ticks it, they expect system titlebars **across the entire application, for all windows**, correct?

It's just not good UX to mix system and client-side decorations. Different look. Possibly different position, if the user places their system decorations on the left, for example. 

So right now the option is there, under Settings, but it doesn't apply to all windows, just to some. Why would anyone want bad UX for Telegram? :) Or am I misunderstanding something?

**Comment by ilya-fedin** [comment]
Your words are useless, I don't decide anything here

**Comment by xmha97** [comment]
> Your words are useless,

I completely agree with @andrei-a-papou.


> I don't decide anything here

Also, if you do not care about our requests, please disable the issue section on GitHub.

**Comment by ilya-fedin** [comment]
> Also, if you do not care about our requests, please disable the issue section on GitHub.

As I already said, I don't decide anything here

**Comment by lokpozzta** [comment]
No not like that but I don’t know how cuz my brother play not me

**Comment by lokpozzta** [comment]
Can’t loin

**Comment by Demetri0** [comment]
The problem is still not solved

**Comment by Nikita-Presnov** [comment]
@john-preston, pay attention to this issue.
A bug with moving windows may pop up.
Not to mention that it is simply ugly when using certain themes, such as in Windows 7.
Purely in my opinion, your csd is not so functional to hold onto it, it does not integrate well into the system, it's just extra work.

**Comment by Demetri0** [comment]
+1 for bug with moving window. (But haven't tested on latest telegram versions)

Also want to add: That current video call window doesn't support such options like: "Always on top" and "Always on visible Workspace" and other really **important** options.

**Comment by ilya-fedin** [comment]
I believe you can enable options like always on top with system-wide hotkeys

**Comment by Demetri0** [comment]
Probably yes, I don't know such hotkeys, will need to google it. But it much easier and comfortable to have this options just by right-click on window frame.

**Comment by ilya-fedin** [comment]
Well, tdesktop has the code to call system window menu, it's just there's no right click handler in the voice chat window.

**Comment by ilya-fedin** [comment]
There's no bug report for that, that's probably why it's not implemented.

**Comment by Demetri0** [comment]
Can I ask why tdesktop team so dislike system window frames? Is there any reason for that?

**Comment by ilya-fedin** [comment]
I don't know exactly, but I guess that's designer's decision: that's how the window looks on the mockup and the developer has to implement the mockup pixel-perfect

**Comment by Demetri0** [comment]
It's really sad when designers doesn't respects user experience, and think only about single application, and not about how this application looks in general system design near with other applications. ;'(

Actually I'm very excited with most tdesktop design, but this decision look really weird for me: I can adjust colors of telegram to be more-less consistent with color scheme of my system, but I can't make frame of window to be comfortable and pretty for me.
Thanks that main window still have this option, I'm afraid at some day and this option may be removed :)

I hope more people vote this issue to give it more weight. (_If it is matter for tdesktop designers_ :))

**Comment by ilya-fedin** [comment]
The content of the window doesn't look like the rest of the system, why the frame should? It has sense for it to look like the content of the window it's drawn around, not the content of other windows.

About the option, yes, @john-preston was saying the option is here only due to bugs in the frame implementation, he'd like to remove the option completely or move it to experimental options at least (as people enable it and then [report design bugs](https://github.com/telegramdesktop/tdesktop/issues/25713#issuecomment-1379896785)). The Windows bug with maximized geometry handling was finally fixed recently, there's also work to use system resize on Windows 11 and Qt changes to fix the event handling on [X11](https://codereview.qt-project.org/c/qt/qtbase/+/471045) and [Wayland](https://codereview.qt-project.org/c/qt/qtwayland/+/467564), so the option indeed could disappear sooner than later.

**Comment by ilya-fedin** [comment]
> (_If it is matter for tdesktop designers_ :))

No, it's not, the only place they read is https://t.me/designers. They're not public people.

**Comment by Demetri0** [comment]
> The content of the window doesn't look like the rest of the system, why the frame should?

Probably you are right, and it is just a luck that content of tdesktop doesn't look alien for my system design.

**Comment by Nikita-Presnov** [comment]
> The content of the window doesn't look like the rest of the system, why the frame should? It has sense for it to look like the content of the window it's drawn around, not the content of other windows.

Totally disagree, and damn it, I want all applications to have a similar design, including the theme, approach to the layout of elements and hotkeys, although this is not possible for the last 2 points.

**Comment by ilya-fedin** [comment]
> I want all applications to have a similar design

Well, you can use Unigram or Telegram for macOS if you're using one of the system they support or create your own client. This is not a goal for tdesktop, sorry. Its goal is to have a uniform cross-platform design.

**Comment by rokejulianlockhart** [comment]
https://github.com/telegramdesktop/tdesktop/issues/26481#issuecomment-1622660724
